project/profiles/capacity4more/modules/c4m/message/c4m_message_digest/c4m_message_digest.install

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * @file
+ * Install and update hooks for the c4m_message_digest module.
+ */
+
+/**
+ * Implements hook_schema().
+ */
+function c4m_message_digest_schema() {
+  $schema['message_digest_preferences'] = [
+    'fields' => [
+      'fid' => [
+        'description' => 'The {flag}.fid of the stored preference.',
+        'type' => 'int',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ],
+      'entity_type' => [
+        'description' => "The entity type (e.g. node, comment, etc').",
+        'type' => 'varchar',
+        'length' => '32',
+        'not null' => TRUE,
+        'default' => '',
+      ],
+      'entity_id' => [
+        'description' => 'The entity id.',
+        'type' => 'int',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ],
+      'uid' => [
+        'description' => 'The {users}.uid of the user.',
+        'type' => 'int',
+        'unsigned' => TRUE,
+        'not null' => TRUE,
+      ],
+      'preference' => [
+        'description' => 'The machine name of the digest preference.',
+        'type' => 'varchar',
+        'length' => 128,
+        'not null' => TRUE,
+        'default' => '',
+      ],
+    ],
+    'indexes' => [
```

Comment:
Shouldn't we add a primary key? Shouldn't we add the entity type as part of the key? Anyway is used in the search so I think the index will not be used.

Reply:
@mfernea do you think there is an advantage in creating a dedicated key column? We will never load / reference a preference by this id.

Reply:
We can add indexes, primary keys or unique keys. It's best to add a composed primary key rather than a composed index. It will ensure uniqueness, which is not enforced at the momment. I would also include the entity_type column in the key for the above reasons.